.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,49 @@
+name: Pull Request Verification
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Setup Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: "3.8"
+
+      - name: Setup Linux
+        if: runner.os == 'linux'
+        run: |
+          sudo apt-get install -y cmake libsndfile1
+
+      - name: Install Ludwig
+        run: pip install ludwig[full]
+
+      - name: Set Ludwig version env
+        run: |
+          echo "ludwig_version=$(python -c "import ludwig; print('.'.join(ludwig.__version__.split('.')[:2]))")" >> $GITHUB_ENV
+
+      - name: Install dependencies
+        run: pip install -r requirements.txt
+
+      - name: Generate API Docs
+        run: python code_doc_autogen.py
+
+      - name: Run Linting
+        run: |
+          pip install pre-commit
+          pre-commit run --all-files
+        continue-on-error: true
```

Comment:
Discussed offline. Main points:

- pre-commit returns an error when it has to modify files. Since code_doc_autogen doesn't produce well-formatted markdown, running pre-commit after code_doc_autogen should result in a clean workspace, but pre-commit itself returns an error.
- The risk of `continue-on-error` hiding non-trivial-file-linting errors is rather low, since we have a separate global pre-commit check. 
- Nevertheless, we scoped out alternative solution, started in #148, where we disable markdownlint for generated markdown files altogether, which also allows this proposed github workflow to be simplified.

Reply:
Absolutely @justinxzhao ! Just rebased and removed the pre-commit block from the github workflow.